File: frontend/src/types/analysis.ts
export interface PossibleCondition {
  name: string;
  reason: string;
}

export interface Reference {
  title: string;
  source: string;
  url: string;
}

export interface AnalysisResponse {
  analysis_id: string;
  image_filename: string;
  symptoms: string;
  status: string;
  message: string;

  image_quality: string;

  visual_observations: string[];
  uncertainties: string[];

  possible_conditions: PossibleCondition[];
  confidence: string | null;

  recommendations: string[];

  seek_professional_help: boolean;

  disclaimer: string;

  references: Reference[];

  skin_score: number;
  skin_level: string;
}

File: backend/app/schemas/analysis.py
from pydantic import BaseModel


class PossibleCondition(BaseModel):
    name: str
    reason: str


class AnalysisResponse(BaseModel):
    analysis_id: str
    image_filename: str
    symptoms: str
    status: str
    message: str

    image_quality: str

    visual_observations: list[str]
    uncertainties: list[str]

    possible_conditions: list[PossibleCondition]
    confidence: str | None

    recommendations: list[str]

    seek_professional_help: bool

    disclaimer: str

    skin_score: int
    skin_level: str